File: front/src/entity/data/LikeResponse.ts
import { Expose } from 'class-transformer'

export class LikeResponse {
  @Expose()
  postId = 0

  @Expose()
  liked = false

  @Expose()
  likesCount = 0

  constructor(postId: number, isLiked: boolean, likesCount: number) {
    this.postId = postId
    this.liked = isLiked
    this.likesCount = likesCount
  }
}

File: src/main/java/com/api/booklog/response/LikeResponse.java
package com.api.booklog.response;

import lombok.Getter;

@Getter
public class LikeResponse {
    private final Long postId;
    private final boolean liked;
    private final Long likesCount;

    public LikeResponse(Long postId, boolean newStatus, Long likesCount) {
        this.postId = postId;
        this.liked = newStatus;
        this.likesCount = (likesCount != null) ? likesCount : 0L;
    }
}
